Drewbranch

ahlopez1999 opened the pull request:
Big merge for implementation of both Water Spread and Erosion. Best way I've found to test and mess around with water spread is to replace the ocean tiles with another base, shape the landscape however you want, then place a single ocean tile and raise it's height to whatever you want to test the spread of water.

ahlopez1999 committed:
updateErosion progress

ahlopez1999 committed:
updateErosion Progress (HUD)

Added HUD display for erosion. Changed rates of erosion based on tile base instead of uniform erosion rate.

ahlopez1999 committed:
UpdateErosion Wokring Implementation

UpdateErosion now in a working state. Exact rate of erosion and what factors should affect it still TBD.

ahlopez1999 committed:
WaterHeight HUD

Adjusted HUD to display the water height on tiles to better test water spread implementation.

ahlopez1999 committed:
Complete Erosion and Comments

Added comments to explain more of my code, and added functionality to what happens when a tile is completely eroded away.

ahlopez1999 committed:
Water Spread Setup

Made a really foolish mistake and forgot to commit my initial changes to updateWater and lost them, so beginning to reestablish them with this commit.

ahlopez1999 committed:
updateWater Progress

ahlopez1999 committed:
Water Spread Progress

ahlopez1999 committed:
Water Spread Tentative

ahlopez1999 committed:
Merge branch 'main' into drewbranch

ahlopez1999 committed:
Merge with Main

Water spread completed and merged with main resolving merge conflicts. Temporarily disabled # of buildings in the HUD display as it was getting too cluttered and running offscreen.

acher0nparth reviewed (changes requested):
Looks good, the ocean stuff especially works nicely I think. I left it running for a bit and it definitely does change the landscape. I do have some comments: 
- The "increase water height" button and "clear water" buttons don't appear to work on your branch, but they do work on main, so I assume their functionality got messed up somewhere?
- The erosion rate seems kind of high. I'm not sure exactly how much land a tile is supposed to represent, but it feels like a whole tile should not be eroding in a year or two's time. This might be something we should base on real-life data, e.g. the SimCoast coastlines ought to erode at roughly the same rate as Florida's coastlines.
- When changing the base of a tile, the erosion percentage of that tile stays the same. I'm not sure if that was intentional, just something to note and to think about. 
- Your water spread functionality is not implementing the "change in water height" variable I added for the building damage system. Without it, the damage calculations won't work properly for tiles with buildings when water spreads onto them. It should be an easy addition.

ahlopez1999 committed:
WaterSpread Fixes

Made water spread work along with Blake's tile damage system, fixed raise water level button, and slowed erosion rates.

acher0nparth reviewed (approved):
Changes look good. Approved!
